### PR TITLE
Update to SF 3.3 and setup Symfony Flex

### PR DESCRIPTION
**Pull request by pyrech**
Build currently fails because of mcrypt deprecation in PHP7.1
Mcrypt is used by `league/oauth2-client` v1.4 and this is fixed in their v2. To make use of their last version we have no choices than updating our slack oauth provider:

- Either PR and get a new release of the current provider `bramdevries/oauth2-slack`;
- or use a new provider (like `adam-paterson/oauth2-slack`).

Our actual provider looks quite inactive and still in 0.10. The alternative is `adam-paterson/oauth2-slack` which looks alive and is recommended by the documentation of the php league. ~It's not yet compatible with `league/oauth2-client` v2 but that should not be too long to get a new release (https://github.com/adam-paterson/oauth2-slack/issues/13)~ the lib is now compatible as of it's 1.1.3 version \o/

Todo:
- [x] update oauth2 slack provider
- [x] update php-cs-fixer to their v2

Note:
- we are now using twig v2 (the only BC break was the use of `_self` variable)

NB: new env var (`APP_ENV`, `APP_DEBUG` and `APP_SECRET`) have been setup on Heroku

### CODE REVIEW

File: etc/routing/app.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+homepage:
```
Comment (damienalexandre) [inline etc/routing/app.yml]: yml, not yaml ?

File: src/SantaKernel.php [hunk 2]
```diff
@@ -1,185 +1,71 @@
...
-     */
-    public function getRootDir()
+    protected function configureContainer(ContainerBuilder $container, LoaderInterface $loader): void
     {
-        if (null === $this->rootDir) {
-            $this->rootDir = dirname(__DIR__);
+        $confDir = dirname(__DIR__) . '/etc';
+        $loader->load($confDir . '/packages/*' . self::CONFIG_EXTS, 'glob');
+        if (is_dir($confDir . '/packages/' . $this->environment)) {
+            $loader->load($confDir . '/packages/' . $this->environment . '/**/*' . self::CONFIG_EXTS, 'glob');
```
Comment (damienalexandre) [inline src/SantaKernel.php]: This looks ugly, is that the standard way?

File: etc/routing/app.yml (hunk 1 above)
Comment (pyrech) [inline etc/routing/app.yml]: My bad

File: src/SantaKernel.php (hunk 2 above)
Comment (pyrech) [inline src/SantaKernel.php]: That allows to override config for a bundle for a given environment ¯\\\_(ツ)\_/¯

**Pull request merged**